apps/admin/frontend/src/components/reporting/ballot_count_report_viewer.tsx

```diff
@@ -0,0 +1,376 @@
+import { Optional, assert, assertDefined } from '@votingworks/basics';
+import { isElectionManagerAuth } from '@votingworks/utils';
+import type { ScannerBatch } from '@votingworks/admin-backend';
+import { LogEventId } from '@votingworks/logging';
+import {
+  getCardCounts,
+  getCastVoteRecordFileMode,
+  getScannerBatches,
+} from '../../api';
+import { AppContext } from '../../contexts/app_context';
+import { PrintButton } from '../print_button';
+import {
+  generateBallotCountReportPdfFilename,
+  generateTitleForReport,
+} from '../../utils/reporting';
+import { ExportReportPdfButton } from './export_report_pdf_button';
+import { FileType } from '../save_frontend_file_modal';
+import { ExportBallotCountReportCsvButton } from './export_ballot_count_report_csv_button';
+
+const ExportActions = styled.div`
+  margin-top: 1rem;
+  margin-bottom: 0.5rem;
+  display: flex;
+  justify-content: start;
+  gap: 1rem;
+`;
+
+const PreviewContainer = styled.div`
+  position: relative;
+  min-height: 11in;
+  margin-top: 0.5rem;
+  padding: 0.5rem;
+  background: rgba(0, 0, 0, 10%);
+  border-radius: 0.5rem;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const PreviewOverlay = styled.div`
+  position: absolute;
+  inset: 0;
+  z-index: 1;
+  background: black;
+  opacity: 0.3;
+`;
+
+const PreviewReportPages = styled.div`
+  section {
+    background: white;
+    position: relative;
+    box-shadow: 0 3px 10px rgb(0, 0, 0, 20%);
+    margin-top: 1rem;
+    margin-bottom: 2rem;
+    width: 8.5in;
+    min-height: 11in;
+    padding: 0.5in;
+  }
+`;
+
+const PreviewActionContainer = styled.div`
+  position: absolute;
+  inset: 0;
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 4rem;
+  display: flex;
+  justify-content: center;
+  align-items: start;
+  z-index: 2;
+`;
+
+const LoadingTextContainer = styled.div`
+  background: white;
+  width: 35rem;
+  border-radius: 0.5rem;
+`;
+
+function Report({
+  electionDefinition,
+  scannerBatches,
+  isOfficialResults,
+  cardCountsList,
+  filter,
+  groupBy,
+  ballotCountBreakdown,
+  generatedAtTime,
+}: {
+  electionDefinition: ElectionDefinition;
+  scannerBatches: ScannerBatch[];
+  isOfficialResults: boolean;
+  cardCountsList: Tabulation.GroupList<Tabulation.CardCounts>;
+  filter: Tabulation.Filter;
+  groupBy: Tabulation.GroupBy;
+  ballotCountBreakdown: Tabulation.BallotCountBreakdown;
+  generatedAtTime: Date;
+}): JSX.Element {
+  const titleGeneration = generateTitleForReport({
+    filter,
+    electionDefinition,
+    scannerBatches,
+    reportType: 'Ballot Count',
+  });
+  const titleWithoutOfficiality = titleGeneration.isOk()
+    ? titleGeneration.ok() ?? `Full Election Ballot Count Report`
+    : 'Custom Filter Ballot Count Report';
+  const title = `${
+    isOfficialResults ? 'Official ' : 'Unofficial '
+  }${titleWithoutOfficiality}`;
+  const customFilter = !titleGeneration.isOk() ? filter : undefined;
+
+  return BallotCountReport({
```

Comment:
haven't seen this render pattern before - is it solving a particular problem? wondering if it ends up bypassing some of React's component lifecycle management in unexpected ways

Reply:
Hm, I didn't notice I was doing anything special in terms of rendering patterns. I think of `Reports` as a locally defined React component, which wraps `BallotCountReport`. It just generates the component, which is either saved as state in the larger component as the preview, or passed to the print methods. I had to abstract the reports component since it has multiple destinations within the `Viewer`. Do you see another way to do it?

Reply:
Ah, I meant that `BallotCountReport` is defined as a component and usually, we'd render that as:
```tsx
return (
  <BallotCountReport
    title={title}
    electionDefinition={electionDefinition}
    // etc
  />
);
```

whereas, this is just calling `BallotCountReport()` directly as a function here - IIRC, there may be some subtle differences and probably not worth breaking pattern unless there's a need to (if for nothing at all, maybe just for code consistency)

Reply:
Oh, I see. Will change it in an upcoming PR where I touch this file!